src/byte_streams.clj

```diff
@@ -6,1019 +6,72 @@
-  [a b]
-  (p/== 0 (compare-bytes a b)))
-
+    [clj-commons.byte-streams]
+    [potemkin.namespaces])
+  (:import (java.nio.file StandardOpenOption)))
+
+;; don't use single-segment potemkin ns to avoid further single-segment issues
+(potemkin.namespaces/import-vars
+  [clj-commons.byte-streams
+
+   conversions
+   inverse-conversions
+   src->dst->transfer
+   byte-array-type
+   seq-of
+   stream-of
+   vector-of
+   type-descriptor
+   normalize-type-descriptor
+   tag-metadata-for
+   def-conversion
+   def-transfer
+   converter
+   seq-converter
+   stream-converter
+   conversion-path
+   convert
+   possible-conversions
+   conversion-cost
+   default-transfer
+   transfer-fn
+   transfer
+   dev-null
+   optimized-transfer?
+   readable-character?
+   print-bytes
+   to-byte-buffer
+   to-byte-buffers
+   to-byte-array
+   to-byte-arrays
+   to-input-stream
+   to-data-input-stream
+   to-output-stream
+   to-char-sequence
+   to-readable-channel
+   to-string
+   to-reader
+   to-line-seq
+   to-byte-source
+   to-byte-sink
+   cmp-bufs
+   compare-bytes
+   bytes=])
 
 
 (comment
-  (require '[manifold.stream :as ms])
-
-  (def content
-    (doto (ms/stream)
-      (ms/put! (clojure.core/byte-array 5))
-      (ms/close!)))
-
-  (conversion-path (stream-of bytes) (seq-of bytes))        ; => ([(stream-of [B) (seq-of [B)])
-
-  (convert content (seq-of bytes))  ; doesn't work
-  (= content (convert content (seq-of bytes)))
-
-  (convert content (seq-of bytes) {:source-type (stream-of bytes)})  ; works
-  (= content (convert content (seq-of bytes) {:source-type (stream-of bytes)}))
-
-  (convert content (seq-of bytes) {:source-type (stream-of nil)})  ; doesn't work
-  (= content (convert content (seq-of bytes) {:source-type (stream-of nil)}))
-
-  (convert content (seq-of String)) ; also doesn't work
-  (= content (convert content (seq-of String)))
-
-  (convert content (seq-of java.nio.ByteBuffer)) ; this works
-  (= content (convert content (seq-of java.nio.ByteBuffer)))
-
-
-
-  (let [content (doto (ms/stream)
-                  (ms/put! (clojure.core/byte-array 5))
-                  (ms/close!))]
-    (convert content (seq-of bytes))  ; no
-    #_ (convert content (seq-of bytes) {:source-type (stream-of bytes)}) ; yes
-    #_(convert content (seq-of bytes) {:source-type (stream-of nil)}) ; no
-    )
-
-
-  (let [content (doto (ms/stream)
-                  (ms/put! (clojure.core/byte-array 5))
-                  (ms/close!))]
-    (convert content (seq-of String)) ; also doesn't work
-    (= content (convert content (seq-of String))))
-
-  (let [content (doto (ms/stream)
-                  (ms/put! (clojure.core/byte-array 5))
-                  (ms/close!))]
-    (convert content (seq-of java.nio.ByteBuffer)) ; this works
-    (= content (convert content (seq-of java.nio.ByteBuffer))))
 
+  ;; compare the namespaces, make sure we got everything
+  (require '[clojure.set :as set])
+
+  (set/difference
```

Comment:
Maybe this can be turned into a Clojure test and the whole comment block can be removed.